{"status": "chatting", "priority": "bug", "title": "Visiting a bad /bugs URL says an error has occurred", "milestone": "later", "superceder": "", "nosylist": ["egor83, jesstess, paulproteus, pythonian4000"], "assigned": "", "waitingon": "", "keywords": [], "id": "474", "files": [], "history": [{"message": "Related to 827?\n   \n", "author": "Coffeeman"}, {"message": "Here is the exception that Roundup raises:\n\nEXCEPTION AT Thu Sep  1 04:40:30 2011\nTraceback (most recent call last):\n  File \"/usr/lib/pymodules/python2.6/roundup/scripts/roundup_server.py\", line\n192, in run_cgi\n    self.inner_run_cgi()\n  File \"/usr/lib/pymodules/python2.6/roundup/scripts/roundup_server.py\", line\n368, in inner_run_cgi\n    tracker.Client(tracker, self, env).main()\n  File \"/usr/lib/pymodules/python2.6/roundup/cgi/client.py\", line 368, in main\n    self.db.close()\n  File \"/usr/lib/pymodules/python2.6/roundup/backends/rdbms_common.py\", line\n1334, in close\n    self.sql_close()\n  File \"/usr/lib/pymodules/python2.6/roundup/backends/back_sqlite.py\", line 307,\nin sql_close\n    self.conn.close()\nOperationalError: unable to close due to unfinalised statements\n\nI added some logging to the Roundup-Django login integration code, and it\ndoesn't seem to be the cause of the problem.\n\nFor now, I will just push this forward to the future. It would be good to be\nfind out what the minimal adjustments to stock Roundup you need to make to be\nable to reproduce this bug. I can't seem to reproduce it on other Roundup\ninstances I find lying around the web.\n   \n", "author": "paulproteus"}, {"message": "egor83, you asked for more clarification on IRC (although I missed you).\n\nWhat I'm trying to say is that /bugs/ is not handled by the main openhatch code\nthat we store on Gitorious at <a href=\"http://gitorious.org/openhatch/oh-mainline\">http://gitorious.org/openhatch/oh-mainline</a> .\n\nIt's handled by Roundup's code, which we happen to have installed on the server\nthrough the Debian package.\n\nSo the bug fix actually would occur in a different codebase than the main\nOpenHatch one, and I know the Roundup codebase less well. That's what I mean by\n\"it's in Roundup\".\n\nSorry that that was unclear. Do my earlier remarks make more sense now?\n   \n", "author": "paulproteus"}, {"message": "It is bitesize, except that it's in Roundup, which is somewhat hard to\nunderstand. I figure that people shouldn't be directed to it as newcomers unless\nthey understand that.\n\nBut if you *do* understand that, then yes!\n   \n", "author": "paulproteus"}, {"message": "I think it's bitesize, isn't it?\n   \n", "author": "egor83"}, {"message": "If I visit <a href=\"http://openhatch.org/bugs/foo\">http://openhatch.org/bugs/foo</a> while not logged in, I get a 400 and\n\n\"Tue Jul 12 19:59:05 2011: An error occurred. Please check the server log for more infomation.\"\n\nVisiting logged in, I get\n\n\"\"\"\nError response\n\nError code 404.\n\nMessage: /bugs/foo.\n\nError code explanation: 404 = Nothing matches the given URI.\n\"\"\"\n\nThey should probably both get a 404, and better yet get a nice styled page for it, a la \n<a href=\"http://openhatch.org/foo.\">http://openhatch.org/foo.</a>\n   \n", "author": "jesstess"}]}